### PR TITLE
fix(openid-connect): openid not closing session and blocking until ttl…

### LINKED ISSUE #8017: openid-connect plugin leaving server-side sessions locked by not calling explicit session:close()
### Current Behavior

Server-side session stores that support locking remain locked after authentication until the default or configured lock ttl ([https://github.com/bungle/lua-resty-session#redis-storage-adapter](https://github.com/bungle/lua-resty-session#:~:text=set%20%24session_redis_maxlockwait%20%20%20%20%20%20%20%20%20%20%20%20%20%2030%3B%20%20%20%23%20(in%20seconds))) expires.  This results in subsequent requests (using the same APISIX session) hanging until the lock expires.

Apisix plugin code: https://github.com/apache/apisix/blob/cf8de8ebc2b00f3fb3db50b3d5415a189c7a0680/apisix/plugins/openid-connect.lua#L312

Does not implement explicit session:close() as recommended by zmartzone/lua-resty-openidc [https://github.com/zmartzone/lua-resty-openidc#sessions-and-locking](https://github.com/zmartzone/lua-resty-openidc#sessions-and-locking:~:text=authenticate(opts)-,session%3Aclose(),-Caching)

### Expected Behavior

APISIX openid-connect plugin should call explicit session:close() and subsequent requests (using the same APISIX session) should complete immediately.

### Error Logs

_No response_

### Steps to Reproduce

1. Configure APISIX openid-connect plugin for a server-side session store that supports locking
2. Configure route using the openid-connect plugin
3. Authenticate on protected route
4. Immediately refresh route using the same session cookie - request will hang until the default or configured lock ttl

### Environment

- APISIX version (run `apisix version`): 2.15.0
- Operating system (run `uname -a`): Linux apisix-6fcbf58999-t8x9r 5.15.0-48-generic # 54-Ubuntu SMP Fri Aug 26 13:26:29 UTC 2022 x86_64 Linux
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`): nginx version: openresty/1.21.4.1
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`): N/A
- APISIX Dashboard version, if relevant: N/A
- Plugin runner version, for issues related to plugin runners: N/A
- LuaRocks version, for installation issues (run `luarocks --version`): /usr/local/bin/luarocks 3.8.

### PR DESCRIPTION
**Pull request by markusmueller**
…l expired when using lockable session storage backend (#8017)

### Description

Fixes #8017

I'm not a Lua developer but I'd really like to see this fixed. It might be appropriate to wrap with a pcall and have the session closed in any case (try-catch-finally). Also, I couldn't quickly figure out if the testing would cover configuring and spinning up a locking session storage (memcache for example) to test this. If so, happy to add a test case.

Thank you for your feedback. 

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by monkeyDluffy6017** [comment]
We need a test case to cover this

**Comment by sheharyaar** [comment]
@markusmueller , can you please add a test case for the PR??

**Comment by markusmueller** [comment]
@sheharyaar I have added a test verifying the session lock is not blocking subsequent requests anymore as the session is properly closed (test will fail without the change in the plugin).

File: apisix/plugins/openid-connect.lua [hunk 1]
```diff
@@ -574,6 +574,9 @@ function _M.rewrite(plugin_conf, ctx)
             end
         end
     end
+    if(session) then
```
Comment (monkeyDluffy6017) [inline apisix/plugins/openid-connect.lua:577]:
```suggestion
    if session then
```

**Review by monkeyDluffy6017 — DISMISSED** [review]

**Comment by markusmueller** [comment]
@monkeyDluffy6017 applied your suggestions, thank you

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by AlinsRan — APPROVED** [review]

**Pull request merged**